### DISCUSSION TITLE
What is the best practive to use Valtio with streaming and suspense

### DISCUSSION THREAD

**Discussion by ivan-angjelkoski** [pmndrs/valtio#708] [Q&A]
lets say i have a `messagesProxy` proxy const, that looks like this: `{messages: []}`.

how can i first fetch the current messages, and then stream new messages to it, using suspense.. what is the best practice?

**Comment by dai-shi** [comment]
Valtio isn't really specifically designed for streaming, but you should be able to do something like this.

```js
const messagesProxy = proxy({ messages: Promise.resolve([]) });
const addMessage = (newMessage) => {
  messagesProxy.messages = messagesProxy.messages.then((prev) => [...prev, newMessage])
};
```
